scipp_widgets/input_spec.py

```diff
@@ -0,0 +1,71 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+# @file
+# @author Matthew Andrew
+import ipywidgets as widgets
+
+
+class InputSpec():
+    """
+    Controls creation and validaton of user-input widgets.
+    """
+    def __init__(self,
+                 name,
+                 validator=lambda input: input,
+                 options=(),
+                 tooltip='',
+                 eval_input=False,
+                 scope={}):
+        """
+        Parameters:
+        name (str): Name of function argument this input corresponds to.
+        validator (Callable[[str], Any]): Validator function.
+        options (List[str]): List of dropdown options.
+        tooltip (str): Widget placeholder text.
+        eval_input (bool): Toggles evaluating input in calling scope.
+        scope (Dict[str: Any]): Non default scope to use for evaluation.
+        """
+        self._name = name
+        self._options = options
+        self._tooltip = tooltip if tooltip else name
+
+        if eval_input:
```

Comment:
So in which cases can I use `even_input=False`, and what is the actual behavior then? Does it only work if the scope is provided?

Reply:
`eval_input=False` is mostly useful when you are inputting strings.

For example If you were specifying a dimension such as `tof`. You can just take the raw input string. With `eval_input=True` you would need to include quotation marks in the input box. 

If we decided to require that string inputs required users to type them in quotes we could drop this entirely and just always evaluate the input.

Reply:
Wouldn't this better be handled by having an new "input" subclass, `StringInput`?

Reply:
That is, if `eval_input` is only there for not messing with things that are actually strings, remove this arg and provide `StringInput` instead?